go/lang/security/audit/database/string-formatted-query.yaml

```diff
@@ -53,117 +53,34 @@ rules:
-        $OBJ.Exec($QUERY, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $QUERY = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.Query($QUERY, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $QUERY = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.ExecContext($CTX, $QUERY, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $QUERY = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryContext($CTX, $QUERY, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $QUERY = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryRow($QUERY)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $QUERY = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryRow($CTX, $QUERY)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $QUERY = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryRowContext($CTX, $QUERY, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.Exec($OTHER, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.Query($OTHER, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.ExecContext($CTX, $OTHER, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryContext($CTX, $OTHER, ...)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryRow($OTHER)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryRow($CTX, $OTHER)
-    - pattern: |
-        $QUERY = "..."
-        ...
-        $OTHER = $FXN(..., $QUERY, ...)
-        ...
-        $OBJ.QueryRowContext($CTX, $OTHER, ...)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.Exec($QUERY, ...)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.Query($QUERY, ...)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.ExecContext($CTX, $QUERY, ...)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.QueryContext($CTX, $QUERY, ...)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.QueryRow($QUERY)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.QueryRow($CTX, $QUERY)
-    - pattern: |
-        $QUERY = $X + ...
-        ...
-        $OBJ.QueryRowContext($CTX, $QUERY, ...)
+    - patterns:
+        - pattern-either:
+            - pattern: $QUERY = fmt.Fprintf($F, "$SQLSTR", ...)
+            - pattern: $QUERY = fmt.Sprintf("$SQLSTR", ...)
+            - pattern: $QUERY = fmt.Printf("$SQLSTR", ...)
+            - pattern: $QUERY = $X + ...
+        - pattern-either:
+            - pattern-inside: |
+                ...
```

Comment:
leading ellipsis statements are *really* slow - this will hurt the performance of this rule a lot. Some kind of upper bound on this ellipsis will improve the performance of this rule a lot - perhaps a function definition? Go is opinionated enough about program structure that this should be an acceptable constraint.

Reply:
Good to know. I tested the lastest commit I pushed and I am getting the expected results. Let me know what you think